src/main/java/com/exactpro/th2/mstore/MessagePersistor.java

```diff
@@ -62,7 +66,7 @@ public MessagePersistor(@NotNull Configuration config, @NotNull CradleStorage cr
         this.futures = new FutureTracker<>();
 
         this.metrics = new MessagePersistorMetrics<>(taskQueue);
-        this.samplerService = Executors.newSingleThreadScheduledExecutor();
+        this.executor = Executors.newScheduledThreadPool(Runtime.getRuntime().availableProcessors(), THREAD_FACTORY); // FIXME: make thread count configurable
```

Comment:
What is the point of making them configurable? They are real threads and it does not make much sense to have them more than available processors (it cluster we manipulate that value via CPU limits)